### PR TITLE
feat(ingestion): add option for optimized skipping of schemas

### PR DESCRIPTION
**Pull request by thomasplarsson**
Fixes: #2208

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by thomasplarsson** [comment]
@hsheth2 This is not the prettiest change I know, but I wanted to just put the PR out there to make any discussion about issue #2208 more concrete. For context, I am stumbling on this when trying to ingest from aws athena where we have in our non-production environment a truly polluted metastore with a ridiculous number of garbage schemas, whereas the number of schemas I would be interested in ingestion from is on the order of <10.

**Review by hsheth2 — CHANGES_REQUESTED** [review]
Overall looks pretty good! Made some minor suggestions

I ran into this same issue when I was ingesting from BigQuery, so I'm glad we've got a good solution for it

File: metadata-ingestion/src/datahub/ingestion/source/sql_common.py [hunk 1]
```diff
@@ -191,44 +196,46 @@ def get_workunits(self) -> Iterable[SqlWorkUnit]:
...
-                    dataset_snapshot.urn = f"urn:li:dataset:(urn:li:dataPlatform:{platform},{dataset_name},{env})"
-                    if description is not None:
-                        dataset_properties = DatasetPropertiesClass(
-                            description=description,
-                            # uri=dataset_name,
-                        )
-                        dataset_snapshot.aspects.append(dataset_properties)
-                    schema_metadata = get_schema_metadata(
-                        self.report, dataset_name, platform, columns
+            if sql_config.schema_pattern.allowed(schema):
```
[the remark this answers is not in the document]
Comment (thomasplarsson) [reply metadata-ingestion/src/datahub/ingestion/source/sql_common.py]: Agree and fixed.

File: metadata-ingestion/src/datahub/ingestion/source/sql_common.py [hunk 2]
```diff
@@ -48,6 +48,11 @@ def report_dropped(self, table_name: str) -> None:
 
 class SQLAlchemyConfig(ConfigModel):
     options: dict = {}
+    # Although the 'table_pattern' enables you to skip everything from certain schemas,
+    # having another option to allow/deny on schema level is an optimization for the case when there is a large number
+    # of schemas that one wants to skip and you want to avoid the time to needlessly fetch those tables only to filter
+    # them out afterwards via the table_pattern.
+    schema_pattern: AllowDenyPattern = AllowDenyPattern.allow_all()
```
[the remark this answers is not in the document]
Comment (thomasplarsson) [reply metadata-ingestion/src/datahub/ingestion/source/sql_common.py:55]: Added.

**Review by hsheth2 — APPROVED** [review]
Looks great! Thanks for adding this.

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**